### PR TITLE
Simplify parsing logic based on python 3.10

### PR DESCRIPTION
**Pull request by marcelveldt**
In preparation of simplifying the datamodels this is part 1, where we no longer have to rely on eval to get the type hints due to the postponed evaluation of types and just use the get_type_hints from the typing package which was not working in python 3.9.

**BREAKING CHANGE:**
This change bumps the minimum python requirement to 3.10 !

There's one eval left (to get back from a string of the cluster/object into the actual class) but that will be removed in a follow up PR where we simplify the datamodel

### CODE REVIEW

File: matter_server/common/helpers/util.py [hunk 1]
```diff
@@ -105,26 +86,20 @@ def parse_utc_timestamp(datetime_string: str) -> datetime:
...
+    if isinstance(value_type, str):
+        value_type = get_type_hints(value_type, globals(), locals())
+
+    # always prefer classes that have a from_dict / FromDict
+    if isinstance(value, dict) and hasattr(value_type, "from_dict"):
+        return value_type.from_dict(value)
+    if isinstance(value, dict) and hasattr(value_type, "FromDict"):
+        return value_type.FromDict(value)
+
     if isinstance(value, dict) and "_type" in value:
```
Comment (balloob) [inline matter_server/common/helpers/util.py:101]: Since this 3 times checks `isinstance`, you can nest the 3 other checks. `isinstance`

**Review by balloob — APPROVED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -8,15 +8,16 @@ version = "2.0.2"
...
 authors = [
     {name = "The Home Assistant Authors", email = "hello@home-assistant.io"}
 ]
 classifiers = [
   "Development Status :: 3 - Alpha",
   "Intended Audience :: Developers",
   "Environment :: Console",
-  "Programming Language :: Python :: 3.9",
   "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
```
[the remark this answers is not in the document]
Comment (balloob) [reply pyproject.toml:20]: I would expect this label to apply to this package, not its dependencies.

[the remark this answers is not in the document]
Comment (flatsiedatsie) [reply pyproject.toml:20]: > It won't work on Python 3.11 until we fix the SDK problems with Python 3.11, right?

Ooh, darn! I've been installing Python 3.11 to anticipate this change :-D Could you share a bit more about this issue? Is it with the Python Matter Server, or with CHIP?

[the remark this answers is not in the document]
Comment (balloob) [reply pyproject.toml:20]: It's the Matter SDK but we a) have a patch and b) are working on fixing it upstream.

[the remark this answers is not in the document]
Comment (MartinHjelmare) [reply pyproject.toml:20]: The label applies to the library as a whole. If its mandatory dependencies don't support a Python version we should not label the library as supporting the Python version.

**Comment by marcelveldt** [comment]
Marked as draft to make it more clear that this one should be merged after the bump of the wheels with the 3.11 patch inside

File: pyproject.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (MartinHjelmare) [reply pyproject.toml:20]: Ok.

**Comment by marcelveldt** [comment]
hmm, test keeps failing on deps, can we somehow point it to a prebuilt wheel for orjson ?

**Pull request merged**